mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -72,14 +72,16 @@ def stop_app(self):
             raise Error('Failed to stop existing apk. Unexpected output: %s' %
                         out)
 
-    def _is_app_installed(self):
+    def check_app_installed(self):
         """Overrides superclass."""
-        try:
-            out = self._adb.shell(
-                'pm list instrumentation | grep ^instrumentation:%s/' %
-                self.package).decode('utf-8')
-            return bool(out)
-        except adb.AdbError as e:
-            if (e.ret_code == 1) and (not e.stdout) and (not e.stderr):
-                return False
-            raise
+        if not self._adb_grep_wrapper(
+            'pm list package | grep %s' % self.package):
+            raise jsonrpc_client_base.AppStartError(
+                '%s is not installed on %s' % (
+                self.app_name, self._adb.getprop('ro.boot.serialno')))
+        if not self._adb_grep_wrapper(
+            'pm list instrumentation | grep ^instrumentation:%s/' %
+            self.package):
```

Comment:
Can you add another grep call after this to make sure the target of the instrumentation is actually installed?